src/service-implementations/lf-repo-tree-node-service.ts

```diff
@@ -350,100 +408,61 @@ export class LfRepoTreeNodeService implements LfTreeNodeService {
-    if (entry.entryType === EntryType.Document) {
-      const document = entry as Document;
-      if (document.extension) {
-        const iconId = IconUtils.getDocumentIconIdFromExtension(document.extension);
-        node.icon = IconUtils.getDocumentIconUrlFromIconId(iconId);
-      }
-      else {
-        node.icon = IconUtils.getDocumentIconUrlFromIconId('document-20');
-      }
-    }
-    else if (entry.entryType === EntryType.Folder) {
-      if (parent?.entryType === EntryType.RecordSeries ||
-        (parent?.entryType === EntryType.Shortcut && parent?.targetType === EntryType.RecordSeries)) {
-        node.icon = IconUtils.getDocumentIconUrlFromIconId('recordfolder-20');
-      }
-      else {
-        node.icon = IconUtils.getDocumentIconUrlFromIconId('folder-20');
-      }
-    }
-    else if (entry.entryType === EntryType.RecordSeries) {
-      node.icon = IconUtils.getDocumentIconUrlFromIconId('recordseries-20');
-    }
-    else if (entry.entryType === EntryType.Shortcut) {
-      const shortcut = entry as Shortcut;
-      // TODO: shortcut doesn't have elecDocumentSize
-      node.targetType = shortcut.targetType;
-      node.targetId = shortcut.targetId;
-      if (shortcut.targetType === EntryType.Document) {
-        const iconId = shortcut.extension ? IconUtils.getDocumentIconIdFromExtension(shortcut.extension) : undefined;
-        const iconUrl = IconUtils.getDocumentIconUrlFromIconId(iconId ?? 'document-20');
-        const shortcutUrl = IconUtils.getDocumentIconUrlFromIconId('shortcut-overlay');
-        node.icon = [iconUrl, shortcutUrl];
-      } else if (shortcut.targetType === EntryType.Folder) {
-        const iconUrl = IconUtils.getDocumentIconUrlFromIconId('folder-20');
-        const shortcutUrl = IconUtils.getDocumentIconUrlFromIconId('shortcut-overlay');
-        node.icon = [iconUrl, shortcutUrl];
-      } else if (shortcut.targetType === EntryType.RecordSeries) {
-        const iconUrl = IconUtils.getDocumentIconUrlFromIconId('recordseries-20');
-        const shortcutUrl = IconUtils.getDocumentIconUrlFromIconId('shortcut-overlay');
-        node.icon = [iconUrl, shortcutUrl];
-      } else {
-        throw new Error('unsupported shortcut targetType');
-      }
-    }
-    else {
-      throw new Error('unsupported entryType');
-    }
-  }
-
-  private setAttributesForEntry(entry: Document | Folder | Shortcut | RecordSeries, node: LfRepoTreeNode) {
+  private setColumnAttributesForEntry(entry: Document | Folder | Shortcut | RecordSeries, node: LfRepoTreeNode) {
     if (this.columnIds) {
       for (const columnId of this.columnIds) {
         node.attributes.set(columnId, this.valueToPropertyValue(entry[columnId], columnId));
       }
     }
   }
 
-  private createFolderNode(entry: Entry, repoName: string, parent?: LfRepoTreeNode): LfRepoTreeNode {
+  private setTargetIdAndTypeForShortcut(entry: Shortcut | Document | Folder | RecordSeries, node: LfRepoTreeNode) {
```

Comment:
entry parameter can just be Entry .. you cast it as a shortcut if entryType is shortcut